app/views/DataView.js

```diff
@@ -40,11 +49,24 @@ module.exports = function (Marionette, Backbone) {
     },
 
     dataCallback: function (packet_name, data) {
-      this.telemetry_data[packet_name] = data;
+      //whether we should overwrite the existing data even if we receive null
+      if (this.keep_last_packet){
+        _.each(this.telemetry_data[packet_name], function(value, header){
```

Comment:
Note to self: its better if this is `data` instead of  this.telemetry_data[packet_name], otherwise might not work if `this.keep_last_packet` is initialized to true. Should test